include/openPMD/RecordComponent.hpp

```diff
@@ -75,6 +75,57 @@ struct IsContiguousContainer< std::array< T_Value, N > >
 };
 } // namespace traits
 
+/**
+ * @brief Subset of C++20 std::span class template.
+ *
+ * Any existing member behaves equivalently to those documented here:
+ * https://en.cppreference.com/w/cpp/container/span
+ */
+template< typename T >
+class Span
+{
+    friend class RecordComponent;
+
+private:
+    using param_t = Parameter< Operation::GET_BUFFER_VIEW >;
+    param_t m_param;
+    size_t m_size;
+    // @todo make this safe
+    Writable * m_writable;
+
+    Span( param_t param, size_t size, Writable * writable ) :
+        m_param( std::move( param ) ),
+        m_size( size ),
+        m_writable( std::move( writable ) )
+    {
+        m_param.update = true;
+    }
+
+public:
+    size_t size() const
+    {
+        return m_size;
+    }
+
+    T *data() const
+    {
+        if( m_param.out->taskSupportedByBackend )
+        {
+            // might need to update
+            m_writable->IOHandler->enqueue( IOTask( m_writable, m_param ) );
+            m_writable->IOHandler->flush();
+        }
+        return static_cast< T * >( m_param.out->ptr );
+    }
+
+    T &operator[]( size_t i ) const
+    {
+        return data()[ i ];
+    }
+};
+
+template class Span< int >;
```

Comment:
Maybe in a `Span.cpp` file?

Reply:
Ah, that one was just for quickly instantiating to get feedback from the compiler. Removed it again.